planner/core/optimizer.go

```diff
@@ -38,6 +38,7 @@ const (
 	flagEliminateAgg
 	flagEliminateProjection
 	flagMaxMinEliminate
+	flagEliminateOuterJoin
```

Comment:
Actually, it's better to put this after predicate push down. Because some parameters will affect elimination. But putting it before PPD cause the following case not to be eliminated.
```
CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL,
  `b` int(11) DEFAULT NULL,
  UNIQUE KEY `a` (`a`)
);
select distinct t1.a from t t1 left join t t2 on t1.a = t2.a where t2.b=1;
```